src/main/java/com/internet/shop/dao/jdbc/UsersDaoJdbcImpl.java

```diff
@@ -0,0 +1,187 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UsersDaoJdbcImpl implements UserDao {
+    @Override
+    public User create(User user) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO users (login, password) VALUES (?, ?);";
+            PreparedStatement statement = connection.prepareStatement(query,
+                    PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getLogin());
+            statement.setString(2, user.getPassword());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            while (generatedKeys.next()) {
+                long userId = generatedKeys.getLong(1);
+                user.setId(userId);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("It is impossible to create a user: "
+                    + user, e);
+        }
+        insertRoleToUsersRolesTable(user);
+        return setRoleToUser(user);
+    }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users WHERE user_id = ? AND deleted = FALSE;";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                User user = createUserFromResultSet(resultSet);
```

Comment:
you are opening a new connection in setRoles method which is inside createUserFromResultSet

Reply:
you can divide  the logic of createUserFromResultSet and setRoles and call them both here(after catch block)